src/lava/magma/core/model/nc/ports.py

```diff
@@ -0,0 +1,358 @@
+# Copyright (C) 2022 Intel Corporation
```

Comment:
Does all the code duplication in this file motivate a refactoring of other classes if this is really necessary?
Perhaps file an issue about that, because clearly that should not remain that way.

Reply:
It's also not quite clear why we would need many of these things in here because clearly we don't need Python implementation of recv() fr a NcProcModel executing on a NeuroCore. 
We only needed that for PyProcModels and corresponding PyPorts executing as Python code on CPU.